### PR TITLE
Make the Prow build URL template human-readable

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @spxtr @kargakis 

Was changing this template downstream in our config and it was a real nuisance to do in one line ... if y'all want this, here is a more readable way to achieve the same thing.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
That config sorting thing does not seem useful

File: prow/config.yaml [hunk 1]
```diff
@@ -1,5 +1,26 @@
 plank:
-  job_url_template: 'https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/{{if eq .Spec.Type "presubmit"}}pr-logs/pull{{else if eq .Spec.Type "batch"}}pr-logs/pull{{else}}logs{{end}}{{if ne .Spec.Refs.Org ""}}{{if ne .Spec.Refs.Org "kubernetes"}}/{{.Spec.Refs.Org}}_{{.Spec.Refs.Repo}}{{else if ne .Spec.Refs.Repo "kubernetes"}}/{{.Spec.Refs.Repo}}{{end}}{{end}}{{if eq .Spec.Type "presubmit"}}/{{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}{{else if eq .Spec.Type "batch"}}/batch{{end}}/{{.Spec.Job}}/{{.Status.BuildID}}/'
+  job_url_template: >-
+    https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/
+      {{- if eq .Spec.Type "presubmit" -}}
```
Comment (spxtr) [inline prow/config.yaml]: I was under the impression that this puts spaces in place of newlines. Is that not right?

Comment (stevekuznetsov) [reply prow/config.yaml]: Hmm -- the one unit test verifying this seems to be OK with it. I'll make a playground

Comment (stevekuznetsov) [reply prow/config.yaml]: Ah sorry -- a little bit of trickery.. this is a YAML folded block so there are no newlines in the string

http://www.yaml.org/spec/1.2/spec.html#id2796251

**Comment by spxtr** [comment]
OK, I trust the test.

That sort test is wrong. I think @krzyzacy might be the right person to fix it?

/assign @krzyzacy

**Comment by krzyzacy** [comment]
cc @yguo0905 
hummm - somewhere around https://github.com/kubernetes/test-infra/blame/master/jobs/config_sort.py#L88-L95, I'm still thinking of a workaround

File: prow/config.yaml [hunk 2]
```diff
@@ -1,5 +1,26 @@
 plank:
-  job_url_template: 'https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/{{if eq .Spec.Type "presubmit"}}pr-logs/pull{{else if eq .Spec.Type "batch"}}pr-logs/pull{{else}}logs{{end}}{{if ne .Spec.Refs.Org ""}}{{if ne .Spec.Refs.Org "kubernetes"}}/{{.Spec.Refs.Org}}_{{.Spec.Refs.Repo}}{{else if ne .Spec.Refs.Repo "kubernetes"}}/{{.Spec.Refs.Repo}}{{end}}{{end}}{{if eq .Spec.Type "presubmit"}}/{{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}{{else if eq .Spec.Type "batch"}}/batch{{end}}/{{.Spec.Job}}/{{.Status.BuildID}}/'
+  job_url_template: >-
```
Comment (krzyzacy) [inline prow/config.yaml]: s/`>-`/`|-`/ use block style seems fine and can still pass the tests?

Comment (yguo0905) [reply prow/config.yaml]: Yeah, if the user of this field can handle newlines.

Comment (stevekuznetsov) [reply prow/config.yaml]: No, newlines are not acceptable

Comment (krzyzacy) [reply prow/config.yaml]: @stevekuznetsov why?

Comment (stevekuznetsov) [reply prow/config.yaml]: Can't have newlines in the URL and the `{{-` syntax in Go templates won't snip them

Comment (stevekuznetsov) [reply prow/config.yaml]: I guess we can post-process this field after we read in config to remove newlines

Comment (krzyzacy) [reply prow/config.yaml]: hummmm - we probably want to have a unit test against the `job_url_template` in config.yaml (thought we had one, but)

Comment (spxtr) [reply prow/config.yaml]: There is a unit test, that's not the problem. The problem is that the config_sort test is trying to sort something that isn't a job.

Comment (krzyzacy) [reply prow/config.yaml]: load and dump is not smart enough :-( it does only *sort* the jobs https://github.com/kubernetes/test-infra/blob/master/jobs/config_sort.py#L90-L92

**Comment by stevekuznetsov** [comment]
This is marginally useful but the infra bits make it more work than it's worth.

/close

**Pull request closed without merging**